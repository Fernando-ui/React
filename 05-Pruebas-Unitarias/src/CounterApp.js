import React,{ useState } from 'react'
import PropTypes from 'prop-types';


const CounterApp = ({value = 0}) => {
    
    const [ counter, setCounter ] = useState(value);
    

    const handleAdd = (e) => {
        setCounter(counter + 1 );
        
    
    }
    const handleReset = (e) => {
        setCounter(value);
        
    
    }
    const handleSubtract = (e) => {
        setCounter(counter - 1 );
        
    
    }
    
    

    return (
        <>
        <h1>Counter App</h1>
        <h2>{counter}</h2>
        <button onClick= {(e) =>{handleSubtract(e)  }}>-1</button>
        <button onClick= {(e) =>{handleReset(e)  }}>Reset</button>
        <button onClick= {(e) =>{handleAdd(e)  }}>+1</button>
        </>
    )

}

export default CounterApp;

CounterApp.propTypes = {
    value:PropTypes.number.isRequired,
}