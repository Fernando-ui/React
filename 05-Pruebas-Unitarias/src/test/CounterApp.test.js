import {shallow} from 'enzyme';
import CounterApp from '../CounterApp';


describe('Pruebas con el conunter App', () => {

    let wrapper = shallow(<CounterApp value={0}/>);

    // para que se reinicie el shallow
    beforeEach(() => {
        wrapper = shallow(<CounterApp value={0}/>);
    })


    test('Debera mostrar <CounterApp /> correctamente  ', () => {
        
        const valor = 10;

        expect(wrapper).toMatchSnapshot();

    });
    test('Debera mostrar <CounterApp /> correctamente con el valor de 100  ', () => {
        
        const valor = 100;
        const wrapper = shallow(<CounterApp value={100} />);
        const valorDeCounter = wrapper.find('h2').text();

        expect(valorDeCounter).toBe('100');

    });

    test('Debe de incrementar con el boton de +1 ', () => {

        wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        console.log(counterText);

        expect(counterText).toBe('1');

    })
    test('Debe de incrementar con el boton de -1 ', () => {
        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        console.log(counterText);

        expect(counterText).toBe('-1');

    })
    test('Debe de incrementar con el boton de +1 ', () => {
        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        console.log(counterText);
        
        expect(counterText).toBe('-1');
        
    })
    
    
    test('Debe de colocar el valro por defecto con el boton reset   ', () => {
        
        const wrapper = shallow(<CounterApp value={105} />);
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('105');
        console.log(counterText);

    })
    
});
