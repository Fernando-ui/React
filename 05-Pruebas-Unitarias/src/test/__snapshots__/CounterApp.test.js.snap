// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el conunter App Debera mostrar <CounterApp /> correctamente   1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;
